Microsoft.Toolkit.Services/Services/LinkedIn/LinkedInDataProvider.cs

```diff
@@ -197,22 +198,18 @@ public async Task<IEnumerable<TSchema>> GetDataAsync<TSchema>(LinkedInDataConfig
 
             var url = $"{_baseUrl}{config.Query}/~:({fields})?oauth2_access_token={Tokens.AccessToken}&format=json&count={maxRecords}&start={startRecord}";
 
-            using (HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri(url)))
-            {
-                request.Headers.Connection.TryParseAdd("Keep-Alive");
-
-                using (var response = await client.SendAsync(request).ConfigureAwait(false))
-                {
-                    var data = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            using HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri(url));
+            request.Headers.Connection.TryParseAdd("Keep-Alive");
 
-                    if (response.IsSuccessStatusCode && !string.IsNullOrEmpty(data))
-                    {
-                        return parser.Parse(data);
-                    }
+            using var response = await client.SendAsync(request).ConfigureAwait(false);
+            var data = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
 
-                    throw new RequestFailedException((System.Net.HttpStatusCode)response.StatusCode, data);
-                }
+            if (response.IsSuccessStatusCode && !string.IsNullOrEmpty(data))
+            {
+                return parser.Parse(data);
```

Comment:
Just a side note for the future (also not sure how much we care about performance here) - especially now that we're using the `System.Text.Json` APIs we might be able to skip the creation of this string entirely (which incurs both the allocation and the internal UTF8 -> Unicode conversion), and just try to deserialize from the raw bytes directly. For instance, if the content size is known, we could rent a `byte[]` array from the pool, copy the content to it, then convert that to a `ReadOnlySpan<byte>` and pass that directly. This is because the `System.Text.Json` APIs actually only work on UTF8 data internally, which means if you deserialize from a `string` they actually rent a `byte[]` array anyway and perform the conversion in the opposite direction to work on that. So here we're essentially doing UTF8 -> Unicode -> UTF8, which is not ideal for both performance (extra steps) and allocation (the wasted `string`).

Again, not a change we need to do now and the current version works just fine.
Just a note I thought I'd share while I was at it 😊

Reply:
The future of this service is uncertain, so I wouldn't bother too much, but yeah, I think I did change a few ReadAsStringAsync to ReadAsStreamAsync.